Branch-SDK/Branch-SDK/BNCLinkCache.m

```diff
@@ -0,0 +1,36 @@
+//
+//  BNCLinkCache.m
+//  Branch-SDK
+//
+//  Created by Qinwei Gong on 1/23/15.
+//  Copyright (c) 2015 Branch Metrics. All rights reserved.
+//
+
+#import "BNCLinkCache.h"
+
+@implementation BNCLinkCache
+
+- (id)init {
+    self = [super init];
+    if (self) {
+        self.cache = [[NSMutableDictionary alloc] init];
+    }
+    return self;
+}
+
+- (void)setObject:(NSString *)anObject forKey:(BNCLinkData *)aKey {
+    [self.cache setObject:anObject forKey:aKey];
+}
+
+- (NSString *)objectForKey:(BNCLinkData *)aKey {
+    NSUInteger hashcode = [aKey hash];
```

Comment:
Something odd with Objective C's NSDictionary. It calls the hash function and returns the same hashcode, but [self.cache objectForKey:aKey] just returns nil